tasks/task_manager.hh

```diff
@@ -126,7 +128,15 @@ public:
             }
 
             virtual future<> abort() noexcept {
-                return make_ready_future<>();
+                if (!_as.abort_requested()) {
+                    _as.request_abort();
+                    auto units = co_await get_units(_children_semaphore, 1);
+                    co_await do_for_each(_children, [] (auto& child) {
```

Comment:
Note, in a coroutine you don't need to co_await do_for_each. It would be simpler to just do:
```c++
for (auto& child : _children) {
    co_await smp::submit_to(child.get_owner_shard(), [&child] () {
        return child->abort();
    });
}
```

That said, there's no reason to do abort the children serially, so co_await `coroutine::parallel_for_each`
rather than `do_for_each`.

Reply:
How about using the get the children task_id:s synchronously in the calling shard and use them to to abort the children on the remote shards, so we can get rid of the _children_semaphore?